dalle2_pytorch/train_diffusion_prior.py

```diff
@@ -0,0 +1,160 @@
+import argparse
+import os
+from dalle2_pytorch import DiffusionPrior
+from embedding_reader import EmbeddingReader
+from dalle2_pytorch import DiffusionPriorNetwork
+import numpy as np
+import math
+from tqdm import tqdm
+import torch
+from torch import nn
+import wandb
+os.environ["WANDB_SILENT"] = "true"
+
+def train(image_embed_dim,
+          image_embed_url,
+          text_embed_url,
+          batch_size,
+          train_percent,
+          val_percent,test_percent,
+          num_epochs,
+          loss_type,
+          condition_on_text_encodings,
+          device,
+          learning_rate=0.01):
+    # DiffusionPriorNetwork 
+    prior_network = DiffusionPriorNetwork( dim = image_embed_dim, depth = 6, dim_head = 64, heads = 8).to(device)
+    
+    # DiffusionPrior with text embeddings and image embeddings pre-computed
+    diffusion_prior = DiffusionPrior( net = prior_network, clip = None, image_embed_dim = image_embed_dim, 
+                                     timesteps = 100, cond_drop_prob = 0.2, loss_type = loss_type,
+                                     condition_on_text_encodings = condition_on_text_encodings).to(device)
+    # Gtext_reader image and text embeddings from the servers
+    print("==============Downloading embeddings - image and text====================")
+    image_reader = EmbeddingReader(embeddings_folder=image_embed_url, file_format="npy")
+    text_reader = EmbeddingReader(embeddings_folder=text_embed_url, file_format="npy")
+
+    ### Training code ###
+    optimizer = torch.optim.SGD(diffusion_prior.parameters(), lr = learning_rate)
```

Comment:
`Adam` is probably better, and given we have a transformer here in the diffusion prior, you'd want to use `AdamW` with some specialized logic to omit the biases

Reply:
https://github.com/lucidrains/tf-bind-transformer/blob/main/tf_bind_transformer/optimizer.py

Reply:
https://github.com/lucidrains/DALLE2-pytorch/blob/main/dalle2_pytorch/optimizer.py#L8 just use this function and you are good to go!